packages/backend/src/config/app.ts

```diff
@@ -123,17 +129,47 @@ function parseRedisTlsConfig(
 
   // self-signed certs.
   if (caFile !== '') {
-    options.ca = readFileSync(caFile)
+    options.ca = fs.readFileSync(caFile)
     options.rejectUnauthorized = false
   }
 
   if (certFile !== '') {
-    options.cert = readFileSync(certFile)
+    options.cert = fs.readFileSync(certFile)
   }
 
   if (keyFile !== '') {
-    options.key = readFileSync(keyFile)
+    options.key = fs.readFileSync(keyFile)
   }
 
   return Object.keys(options).length > 0 ? options : undefined
 }
+
+// exported for testing
+export function parseOrProvisionKey(keyPath: string): {
+  key: crypto.KeyObject
+  jwk: crypto.JsonWebKey
+} {
+  try {
+    const key = crypto.createPrivateKey(fs.readFileSync(keyPath))
+    const jwk = key.export({ format: 'jwk' })
+    if (jwk.crv === 'Ed25519') {
+      return { key, jwk }
+    } else {
+      console.log('Private key is not EdDSA-Ed25519 key. Generating new key.')
+    }
+  } catch {
+    console.log('Private key could not be loaded. Generating new key.')
+  }
+  const keypair = crypto.generateKeyPairSync('ed25519')
+  if (!fs.existsSync(TMP_DIR)) {
+    fs.mkdirSync(TMP_DIR)
+  }
+  fs.writeFileSync(
+    PRIVATE_KEY_FILE,
```

Comment:
Now I'm wondering if `parseOrProvisionKey` should only operate with `keyPath` as opposed to `TMP_DIR`/`PRIVATE_KEY_FILE`, to prevent an operator from not realizing that the default `PRIVATE_KEY_PATH` is being used and not their configured one (even though that is being logged).

Reply:
I'm not sure I understand. `PRIVATE_KEY_FILE` does include the path.

Reply:
As in `parseOrProvisionKey` would only read or write (maybe not overwrite) the file at `keyPath`.
As is, it reads from `keyPath` but writes to `PRIVATE_KEY_FILE`.

Reply:
But what if it only has read access and not write access at `keyPath`?

Reply:
I'm imagining an operating running multiple instances of rafiki `backend` in a k8s cluster and configuring `PRIVATE_KEY_FILE` as a path to a secret.
In which case, I think we should throw if we can't use (or write to) the configured file (`keyPath`).
Otherwise you end up with individual instances provisioning their own private keys.